### PR TITLE
fix(skip-link,region): Allow multiple skiplinks at page top

### PR DESCRIPTION
**Pull request by straker**
Redefine a skip link as any anchor element whose `href` starts with `#...` and which precedes the first anchor element whose `href` doesn't start with `#..`. (that is, a link to a page)

Closes: #1346
Closes: #863

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**
- [x] Follows the commit message policy, appropriate for next version
- [x] Has documentation updated, a DU ticket, or requires no documentation change
- [x] Includes new tests, or was unnecessary
- [x] Code is reviewed for security by: @WilcoFiers

### CODE REVIEW

**Comment by straker** [comment]
We should hold off merging this for just a bit. As it stands, it increases the time of the `skip-link` rule on a page of >11k elements from ~30ms to ~140ms. We should figure out a way to cache the `firstPageLink` so we only have to do it once for the entire run instead of for each anchor element.

File: lib/commons/dom/is-skip-link.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	}
+
+	// define a skip link as any anchor element whose href starts with `#...`
+	// and which precedes the first anchor element whose href doesn't start
+	// with  `#...` (that is, a link to a page)
+	const firstPageLink = axe.utils.querySelectorAll(
+		axe._tree,
+		'a:not([href^="#"]):not([href^="/#"]):not([href^="javascript"])'
+	)[0];
+	if (firstPageLink) {
```
Comment (WilcoFiers) [inline lib/commons/dom/is-skip-link.js]: Nitpick: In the spirit of keeping the [happy path on the left](https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88), I'd suggest `if (!firstPageLink) { return true }`. Not having a first link is an exception, not the default.

File: lib/commons/dom/is-skip-link.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+/**
+ * Determines if element is a skip link
+ * @method isSkipLink
+ * @memberof axe.commons.dom
+ * @instance
+ * @param  {Element} element
+ * @return {Boolean}
+ */
+dom.isSkipLink = function(element) {
```
Comment (WilcoFiers) [inline lib/commons/dom/is-skip-link.js]: We'll need to have tests for this. Feel free to move some of the tests of the check into a separate tests for this commons.

**Comment by straker** [comment]
When #1503 is merged, I can cache the firstPageLink so we won't have to recalculate it for each node.

File: test/checks/navigation/skip-link.js [hunk 3]
```diff
@@ -7,31 +7,26 @@ describe('skip-link', function() {
...
-		assert.isFalse(checks['skip-link'].evaluate(node));
-	});
-
 	it('should return true if the href points to an element with an ID', function() {
 		fixture.innerHTML =
 			'<a href="#target">Click Here</a><h1 id="target">Introduction</h1>';
+		axe._tree = axe.utils.getFlattenedTree(fixture);
 		var node = fixture.querySelector('a');
-		assert.isTrue(checks['skip-link'].evaluate(node));
+		assert.isTrue(axe.commons.dom.isSkipLink(node));
```
Comment (WilcoFiers) [inline test/checks/navigation/skip-link.js]: Why are you running a commons here, and not the check?

File: test/checks/navigation/skip-link.js [hunk 4]
```diff
@@ -7,31 +7,26 @@ describe('skip-link', function() {
...
 	});
 
 	it('should return true if the href points to an element with an name', function() {
 		fixture.innerHTML = '<a href="#target">Click Here</a><a name="target"></a>';
+		axe._tree = axe.utils.getFlattenedTree(fixture);
 		var node = fixture.querySelector('a');
-		assert.isTrue(checks['skip-link'].evaluate(node));
+		assert.isTrue(axe.commons.dom.isSkipLink(node));
```
Comment (WilcoFiers) [inline test/checks/navigation/skip-link.js]: As above.

File: test/checks/navigation/skip-link.js (hunk 3 above)
Comment (straker) [reply test/checks/navigation/skip-link.js]: I think I had moved them to the `isSkipLink` test before deciding to move it back, but forgot to change it back to a check test. Good catch.

**Pull request merged**